ray-on-gke/tpu/kuberay-tpu-webhook/main.go

```diff
@@ -635,6 +607,41 @@ func init() {
 func main() {
 	flag.Parse()
 
+	// use in-cluster config if kubeConfig path is not passed as a flag
+	if KubeConfigPath == "" {
+		config, err := rest.InClusterConfig()
+		if err != nil {
+			panic(err)
+		}
+		client = kubernetes.NewForConfigOrDie(config)
+	} else {
+		config, err := clientcmd.BuildConfigFromFlags("", KubeConfigPath)
+		if err != nil {
+			panic(err)
+		}
+		client = kubernetes.NewForConfigOrDie(config)
+	}
+
+	// instantiate PodInformer for Ray worker pods in the GKE cluster
+	tweakListOptionsFunc := func(options *metav1.ListOptions) {
+		options.LabelSelector = "ray.io/node-type=worker"
+	}
+	factory := informers.NewFilteredSharedInformerFactory(client, 5*time.Minute, metav1.NamespaceAll, tweakListOptionsFunc)
+	podInformer = factory.Core().V1().Pods()
+
+	// Log the intercepted Pod
+	podInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			pod := obj.(*corev1.Pod)
+			klog.V(1).InfoS("Intercepted Pod", "Namespace", pod.Namespace, "Name", pod.Name)
```

Comment:
this log message is kind of misleading, the event heandler is not intercepting any Pod requests, it's just receiving an event for the Pod

Reply:
I'll update the log message to reflect that. Unrelated, but for some reason defining the `AddFunc` Event here is necessary for the PodInformer to cache the Pods before the webhook intercepts them. From what I could find online, this shouldn't be necessary (i.e. Pods should be added to the cache in real time), but in my testing without this piece of logic the PodInformer missed the Pods and always returned an empty list.